jwst/extract_1d/soss_extract/soss_extract.py

```diff
@@ -886,7 +924,19 @@ def model_image(
 
         # Run the extract method of the Engine.
         log.info("Running extraction engine for overlapping orders 1 & 2...")
-        f_k = engine(self.scidata_bkg, self.scierr, tikhonov=True, factor=tikfacs_out["Order 1"])
+
+        if resources is not None and resources.Minv is None:
+            _Minv, _bmat, _mask = engine.return_matrices(self.scidata_bkg, self.scierr, tikfac=tikfacs_out["Order 1"])
+            resources.Minv = _Minv
+            resources.bmat = _bmat
+            resources.mask = _mask
+
+        if resources is not None:
+            y_over_err = (self.scidata_bkg/resources.mederr)[~resources.mask]
+            f_k = resources.Minv.dot(resources.bmat.T*y_over_err)
+        else:
+            f_k = engine(self.scidata_bkg, self.scierr,
+                         tikhonov=True, factor=tikfacs_out["Order 1"])
```

Comment:
Can you help me understand the logic here?  Is the idea that on the first integration you'll initialize a `SharedResources` object, but it'll have `Minv` and `bmat` still set to None?  And then in that first integration these will be computed?

Reply:
The intent is that if resources is None the code is the same as before to provide for a natural option for the old behavior.  Is resources is not None and it already has the required matrices, use them.  If resources is not None and it doesn't have the required matrices, then compute them.  I agree that this could be clearer; I will add a comment.